### PR TITLE
develop-boxed -> develop

### PR DESCRIPTION
**Pull request by heifner**
## Change Type
**Select *ONE*:**
- [ ] Documentation

- [ ] Stability bug fix

- [x] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Comment by swatanabe-b1** [comment]
Is there a plan to handle the rodeos kv intrinsics?  Rodeos is basically broken until that is solved.

File: libraries/chain/include/eosio/chain/webassembly/interface.hpp [hunk 1]
```diff
@@ -2,12 +2,57 @@
...
+     constexpr host_function_registrator(Mod mod_name, Name fn_name) {
+         using rhf_t = eos_vm_host_functions_t;
+         rhf_t::add<HostFunction, Preconditions...>(mod_name.c_str(), fn_name.c_str());
+#ifdef EOSIO_EOS_VM_OC_RUNTIME_ENABLED
+         eosvmoc::register_eosvm_oc<HostFunction, std::tuple<Preconditions...>>(mod_name + BOOST_HANA_STRING(".") + fn_name);
+#endif
+      }
+   };
+
+#define REGISTER_HOST_FUNCTION(NAME, ...) \
```
Comment (swatanabe-b1) [inline libraries/chain/include/eosio/chain/webassembly/interface.hpp:27]: You should take the version that puts the host function registration in eos-vm.cpp.  This change was made to reduce compile times.

Comment (heifner) [reply libraries/chain/include/eosio/chain/webassembly/interface.hpp]: Thanks. Moved.

File: libraries/chain/include/eosio/chain/webassembly/interface.hpp [hunk 2]
```diff
@@ -306,273 +304,35 @@ namespace webassembly {
...
-         double _eosio_ui64_to_f64(uint64_t) const;
-
-         /**
-          * Get the list of active producer names.
-          *
-          * @ingroup producer
-          * @param[out] producers - output buffer containing the names of the current active producer names.
-          *
-          * @return number of bytes required (if the buffer is empty), or the number of bytes written to the buffer.
-         */
```
Comment (swatanabe-b1) [inline libraries/chain/include/eosio/chain/webassembly/interface.hpp:376]: I think you should keep these comments.  It'll make merging easier in the future.

File: libraries/chain/include/eosio/chain/webassembly/eos-vm-oc.hpp [hunk 3]
```diff
@@ -42,225 +58,14 @@ class eosvmoc_runtime : public eosio::chain::wasm_runtime_interface {
...
-   uint64_t scratch;
-
-   asm volatile("mov %%gs:(%[Ptr]), %[Dumpster]\n"                   //probe memory location at ptr to see if valid
-                "mov %%gs:%c[firstInvalidMemory], %[Scratch]\n"      //get first invalid memory address
-                "cmpb $0, %%gs:-1(%[Scratch])\n"                     //is last byte in valid linear memory 0?
-                "je 2f\n"                                            //if so, this will be a null terminated string one way or another
-                "mov %[Ptr],%[Scratch]\n"
-                "1:\n"                                               //start loop looking for either 0, or until we SEGV
-                "inc %[Scratch]\n"
-                "cmpb $0,%%gs:-1(%[Scratch])\n"
```
Comment (swatanabe-b1) [inline libraries/chain/include/eosio/chain/webassembly/eos-vm-oc.hpp:91]: This fix needs to be preserved.

File: libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/intrinsic_interface.hpp [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+   uint64_t scratch;
+
+   asm volatile("mov %%gs:(%[Ptr]), %[Dumpster]\n"                   //probe memory location at ptr to see if valid
+                "mov %%gs:%c[firstInvalidMemory], %[Scratch]\n"      //get first invalid memory address
+                "cmpb $0, %%gs:-1(%[Scratch])\n"                     //is last byte in valid linear memory 0?
+                "je 2f\n"                                            //if so, this will be a null terminated string one way or another
+                "mov %[Ptr],%[Scratch]\n"
+                "1:\n"                                               //start loop looking for either 0, or until we SEGV
+                "inc %[Scratch]\n"
+                "cmpb   $0,%%gs:(%[Scratch])\n"
```
Comment (swatanabe-b1) [inline libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/intrinsic_interface.hpp:55]: Fix from eos-vm-oc.hpp needed here.

**Review by swatanabe-b1 — CHANGES_REQUESTED** [review]

File: libraries/chain/include/eosio/chain/webassembly/interface.hpp (hunk 2 above)
Comment (heifner) [reply libraries/chain/include/eosio/chain/webassembly/interface.hpp:376]: Good find. Added back.

File: libraries/chain/include/eosio/chain/webassembly/eos-vm-oc.hpp (hunk 3 above)
Comment (heifner) [reply libraries/chain/include/eosio/chain/webassembly/eos-vm-oc.hpp:91]: Hmm, scary that slipped through. I was actually going to look for that and forgot. Great catch!!

File: libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/intrinsic_interface.hpp (hunk 4 above)
Comment (heifner) [reply libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/intrinsic_interface.hpp:55]: Added.

**Review by swatanabe-b1 — APPROVED** [review]

**Pull request merged**